modules/transport/http/src/org/apache/axis2/transport/http/util/HTTPProxyConfigurationUtil.java

```diff
@@ -154,6 +156,20 @@ public static void configure(MessageContext messageContext,
             proxyPort = Integer.parseInt(port);
         }
 
+        String userName = System.getProperty(HTTP_PROXY_USER);
+        if (userName != null) {
+            proxyUser = userName;
+        }
+
+        String password = System.getProperty(HTTP_PROXY_PASSWORD);
+        if (password != null) {
+            proxyPassword = password;
+        }
+
+        if (proxyUser != null && proxyPassword != null && proxyCredentials == null) {
```

Comment:
I think you have added proxyCredentials == null check to prevent NTCredentials getting replaced. That leads to a inconsistent behaviour. If we say we can override credentials by system properties it has to happen for all the cases. So taking the domain from a system property and creating NTCredentials instance is much more appropriate here.